### ISSUE TITLE
Reading from a file seems rather unlike existing prior art like `fetch`?

### ISSUE THREAD

**Issue by benlesh** [WICG/file-system-access#433]
While I understand that `fetch` generally has fewer ways of getting http responses than something dealing with opening and reading files, this API -- with regards to reading files in particular -- is surprisingly divergent, IMO. I personally found the initial example to be a bit hard to follow, where I was bouncing around inside the code to figure out what was being read, what happens when it's read, and what's initializing the read.

I guess I would have expected something like `fetch` and how you read the `body`:

```ts
const response = await fetch('/some/image.png');

if (!response.ok) {
  throw new Error('Not good');
}

const reader = await response.body.getReader();

for await (const data of reader) {
  // do things with binary data here.
}
```

Where a file read that aligns more with the existing API might look like:

```ts
const files = await showOpenFilePicker(/* options */);

for (const file of files) {
  console.log(`Reading ${file.name}`);
  
  // Get a reader
  const fileReader = await file.content.getReader();
  
  // Read the binary data
  for await (const data of fileReader) {
    // do something with binary data here
  }
}
```

**Comment by a-sully** [comment]
One thing I'll mention before addressing your feedback...

> I personally found the initial example to be a bit hard to follow, where I was bouncing around inside the code to figure out what was being read, what happens when it's read, and what's initializing the read.

The `getFile()` method is specified over in https://github.com/whatwg/fs ([here](https://fs.spec.whatwg.org/#api-filesystemfilehandle-getfile)). So
- if things are hard to follow.... yeah sorry about that. I wish things weren't split across two specs, too
- if you have feedback about that method in particular, please consider filing an issue on that spec (and we can close this one)

> While I understand that `fetch` generally has fewer ways of getting http responses than something dealing with opening and reading files, this API -- with regards to reading files in particular -- is surprisingly divergent, IMO. 

The prior art this API builds on is [`FileSystemFileEntry.file()`](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemFileEntry/file) which returns (via a callback, since it predates promises) a [`File`](https://developer.mozilla.org/en-US/docs/Web/API/File). `File` extends `Blob`, from which you can (among other things) get a `ReadableStream` (which is what `response.body` gets you)

```
const readableStream = (await fileHandle.getFile()).stream()
```
